Engine/ScpiInstrument.cs

```diff
@@ -369,11 +376,14 @@ public override void Open()
                         base.Open();
                         SetTerminationCharacter(scpiIO.ID);
 
-                        IdnString = QueryIdn();
-                        IdnString = IdnString.Trim();
-                        Log.Info("Now connected to: " + IdnString);
+                        if (SendIDNOnConnect)
+                        {
+                            IdnString = QueryIdn();
+                            IdnString = IdnString.Trim();
+                            Log.Info("Now connected to: " + IdnString);
 
-                        CommandCls(); // Empty error log
+                            CommandCls(); // Empty error log
+                        }
```

Comment:
Actually I think CommandCls should be outside the if?
```suggestion

                        }
                        CommandCls(); // Empty error log
```

Reply:
Good suggestion, I added a new boolean for SendCLSOnConnect, more flexibility for the user